### PR TITLE
Fix #5440: Change ProfileId defaulting to be an invalid profile

### LINKED ISSUE #5440: [Feature Request]: Change ProfileId defaulting to be an invalid profile
### Is your feature request related to a problem? Please describe.
Currently, the proto3 `ProfileId` message suffers from a critical design flaw that creates potential bugs and inconsistencies:

#### Current Issue
- **Default proto3 behavior**: Unspecified `ProfileId` messages default integers to `0`
- **Admin profile initialization**: The admin profile (first profile created) is also assigned ID `0`
- **Logged out state**: Uses `-1` as the default logged-out ID
- **Inconsistency problem**: Code may assume a default `ProfileId` of `0` means "invalid/uninitialized" when it actually corresponds to the admin profile

This creates a dangerous ambiguity where uninitialized `ProfileId` objects are indistinguishable from valid admin profile references.

### Proto Schema Refactoring
Introduce a proper uninitialized state by restructuring `ProfileId` with an optional internal profile ID:

```proto
// Rename of the existing proto.
message LegacyProfileId {
  int32 internal_id = 1;
}

message ProfileId {
  optional int32 internal_id = 1;
}

...

// Updated Profile proto.
message Proto {
  LegacyProfileId legacy_id = 1;
  ProfileId id = 23; // To handle migrations.
```

#### Required Codebase Changes
To make this refactor easy to follow and contribute to, the following steps should be consistently applied in every PR that updates a package:

1. Handle both logged-out and uninitialized states explicitly
   - ProfileId should never rely on default 0 to determine state.
   - For logged-out state, use the default proto (i.e. invalid ID means the same as logged out).

2. Audit all ProfileId usage & extraction points
   - Replace all usages of internalProfileId or profileId.internalId that assume the field is always populated.
   - Before accessing internal_id, check whether it's available:

```kotlin
if (profileId.hasInternalId()) {
  // Safe to use internalId.
}
```

Note that this will require a migration in places where profile ID is used today persistently since we want to move all profile IDs to be +1:
- `ProfileManagementController` needs to add +1 to all profile IDs not using the new structure, and start `nextProfileId` at 1 for new installations.
- `PersistentCacheStore` will need to be migrated by shifting all of the per-profile cache store directories to be +1.
- `EventBundleCreator` needs to be updated to use new field names for the new profile ID (`profileId2`).

### How to Take on This Issue
If you want to contribute to this refactor, please:

1. **Pick a specific package** from the checklist below
2. **Leave a comment** on this thread that includes:
   - Which package you're claiming (e.g., `org.oppia.android.app.profile`)
   - A brief explanation of how you'll handle the `ProfileId` refactoring in that package
   - Mention any specific files or classes in that package that use `ProfileId`
   - Show understanding of the oneof pattern by explaining how you'll replace direct `internalId` access with proper type checking

This helps maintainers verify that you understand the requirements and prevents duplicate work.

### When Creating Your PR
Each PR should include:

- **Package-specific `ProfileId` usage updates** - Replace all direct access to `internalId` with proper oneof type checking.
- **Unit test updates** - Ensure tests cover both valid and invalid profile IDs.

### PR Guidelines
- **One PR per package** to maintain manageable review scope
- **Reference this issue** in your PR description
- **Include before/after examples** of key changes in your PR description
- **Test thoroughly** - Pay special attention to edge cases involving uninitialized ProfileIds

#### Packages Requiring Updates
- [ ] `org.oppia.android.app.profile`
- [ ] `org.oppia.android.app.administrationcontrols`
- [ ] `org.oppia.android.app.classroom`
- [ ] `org.oppia.android.app.completedstorylist`
- [ ] `org.oppia.android.app.drawer`
- [ ] `org.oppia.android.app.help`
- [ ] `org.oppia.android.app.home`
- [ ] `org.oppia.android.app.home.promotedlist`
- [ ] `org.oppia.android.app.home.recentlyplayed`
- [ ] `org.oppia.android.app.onboarding`
- [ ] `org.oppia.android.app.options`
- [ ] `org.oppia.android.app.player`
- [ ] `org.oppia.android.app.profile`
- [ ] `org.oppia.android.app.profileprogress`
- [ ] `org.oppia.android.app.resumelesson`
- [ ] `org.oppia.android.app.settings`
- [ ] `org.oppia.android.app.shim`
- [ ] `org.oppia.android.app.spotlight`
- [ ] `org.oppia.android.app.splash`
- [ ] `org.oppia.android.app.story`
- [ ] `org.oppia.android.app.survey`
- [ ] `org.oppia.android.app.testing`
- [ ] `org.oppia.android.app.topic`
- [ ] `org.oppia.android.app.topic.revisioncard`
- [ ] `org.oppia.android.app.walkthrough`
- [ ] `org.oppia.android.domain.devoptions`
- [ ] `org.oppia.android.domain.exploration`
- [ ] `org.oppia.android.domain.question`
- [ ] `org.oppia.android.testing.lightweightcheckpointing`
- [ ] `org.oppia.android.util.logging`
- [ ] `org.oppia.android.util.profile`

### PR DESCRIPTION
**Pull request by subhajitxyz**
## Explanation
Fix #5440 : Change ProfileId defaulting to be an invalid profile

- changed profile id proto.
- changed use of profileid in org.oppia.android.app.player.state

## Essential Checklist

- [ ] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [ ] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [ ] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

File: app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt [hunk 1]
```diff
@@ -307,10 +310,17 @@ class StateFragmentPresenter @Inject constructor(
...
+    when (profileId.typeCase) {
+      ProfileId.TypeCase.INTERNAL_ID -> {
+        if (getAudioFragment() == null) {
+          val audioFragment: AudioFragment = AudioFragment.newInstance(profileId.internalId)
+          fragment.childFragmentManager.beginTransaction()
+            .add(R.id.audio_fragment_placeholder, audioFragment, TAG_AUDIO_FRAGMENT).commitNow()
+        }
+      }
+      ProfileId.TypeCase.LOGGED_OUT,
+      ProfileId.TypeCase.TYPE_NOT_SET -> return null
```
Comment (subhajitxyz) [inline app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt:322]: Hi @adhiamboperes  — what is the expected behaviour when ProfileId is LOGGED_OUT or TYPE_NOT_SET?

Comment (adhiamboperes) [reply app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt:322]: An error should always be logged in such instances.

**Comment by adhiamboperes** [comment]
@subhajitxyz, will there be an update for this PR?
